LICENSE

```diff
@@ -328,3 +329,68 @@ As between Initial Developer and the Contributors, each party is responsible for
+are permitted provided that the following conditions are met:
+
+ * Redistributions of source code must retain the above copyright notice, this
+   list of conditions and the following disclaimer.
+
+ * Redistributions in binary form must reproduce the above copyright notice,
+   this list of conditions and the following disclaimer in the documentation
+   and/or other materials provided with the distribution.
+
+ * Neither the name Facebook nor the names of its contributors may be used to
+   endorse or promote products derived from this software without specific
+   prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+------------------------------------------------------------------------------------
+This distribution has a binary dependency on zstd-jni, which is available under the BSD 2-Clause License
+as described below.
+
+Zstd-jni: JNI bindings to Zstd Library
+
+Copyright (c) 2015-2016, Luben Karavelov/ All rights reserved.
+
+BSD License
+
+Redistribution and use in source and binary forms, with or without modification,
+are permitted provided that the following conditions are met:
+
+* Redistributions of source code must retain the above copyright notice, this
+  list of conditions and the following disclaimer.
+
+* Redistributions in binary form must reproduce the above copyright notice, this
+  list of conditions and the following disclaimer in the documentation and/or
+  other materials provided with the distribution.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
```

Comment:
Is this LICENSE change needed? Cc @ewencp

Reply:
I'm not sure why most of the additions here were made, tbh.

All of this with the caveat IANAL and any questions re: licensing are best directed at ASF legal.

In many cases, the NOTICE file might be a more appropriate location depending on where the LICENSE actually gets pulled into the binary distribution. This gets confusing because of handling both source and binary distributions. In source distributions, including additional info in the LICENSE file would really only apply to anything *copied* in afaik, since the source distribution doesn't include any of the dependency binaries, i.e. what's under license in that distribution is only our code. (Probably gets messier with top-of-file copyright notices..., and even NOTICE files seem messy because, e.g., Apache explicitly calls them out while other licenses don't)

In binary distributions, adding here wouldn't change, for example, whether we *reproduce* the BSD license notice as requested -- either you think it needs to be directly printed (which this doesn't do) or just including the original JARs with their original licenses would be enough since any binary distribution that *includes* the dependencies would have that in the jar (or if they were missing, upstream should fix that). At a bare minimum, we don't currently include nearly all the dependency licenses here and I don't think most Apache projects do. I haven't looked back at the history to understand what triggered people to start including more than the Apache license in this file.

Again, IANAL, but it's actually unclear you *really* need anything beyond the raw license given the way source distributions and java binary distributions would work & preserve other contents (modulo uberjar conflicts, which we don't do). I think Apache is the only OSS license where there's a "readable" phrase that might complicate things.

But IANAL, so probably figuring this out w/ apache legal would give us more confidence in the answer.

Reply:
@ijuma @ewencp The spark folks appear to have taken a principled approach here: https://github.com/apache/spark/pull/21640. Ultimately they still include the zstd and zstd-jni license files. I think we can probably commit this as is and perhaps file a JIRA to come up with a standardized approach (probably just use the same convention as spark). Does that sound reasonable?

Reply:
FYI: In fact, apache/spark#21640 (what @hachikuji pointed out) is what I referred to while I was working on this commit. (It is also mentioned in the [KIP document](https://cwiki.apache.org/confluence/display/KAFKA/KIP-110%3A+Add+Codec+for+ZStandard+Compression#KIP-110:AddCodecforZStandardCompression-License).) It was initiated from [this message in spark-dev mailing list](http://mail-archives.apache.org/mod_mbox/spark-dev/201806.mbox/%3C4D5001D8-D24A-4FC9-A942-50616A8EB13B@classsoftware.com%3E), sent by a PMC member.

Reply:
@hachikuji I'm fine with merging and submitting a JIRA. It's clear that we're not consistent, so we probably should clean it up before the release.

Reply:
Sounds good. I will merge today.